tools/issue-labeler/labeler/github.go

```diff
@@ -0,0 +1,78 @@
+package labeler
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+)
+
+func newGitHubClient() *github.Client {
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: os.Getenv("GITHUB_TOKEN")},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+	return github.NewClient(tc)
+}
+
+// Helper functions
+func splitRepository(repository string) (string, string, error) {
+	var owner, repo string
+	or := strings.Split(repository, "/")
+	if len(or) != 2 {
+		return "", "", fmt.Errorf("unexpected repository format %s", repository)
+	}
+
+	owner = or[0]
+	repo = or[1]
+	return owner, repo, nil
+}
+
+func ensureLabelWithColor(client *github.Client, owner, repo, labelName, color string) error {
+	existingLabel, err := getLabel(client, owner, repo, labelName)
```

Comment:
Rather than get each service label, what would you think about listing all the labels up front using https://docs.github.com/en/rest/issues/labels?apiVersion=2022-11-28#list-labels-for-a-repository?

That would enable a function that takes the list of labels in the file vs the API and computes the labels that need to be edited / created, which could be unit tested, and then the outputs could be used to make the actual API calls.

Reply:
sounds reasonable